src/diffusers/pipelines/stable_diffusion/__init__.py

```diff
@@ -27,6 +29,32 @@ class StableDiffusionPipelineOutput(BaseOutput):
     nsfw_content_detected: List[bool]
 
 
+@flax.struct.dataclass
+class FlaxStableDiffusionPipelineOutput(BaseOutput):
+    """
+    Output class for Stable Diffusion pipelines.
+
+    Args:
+        images (`List[PIL.Image.Image]` or `np.ndarray`)
+            List of denoised PIL images of length `batch_size` or numpy array of shape `(batch_size, height, width,
+            num_channels)`. PIL images or numpy array present the denoised images of the diffusion pipeline.
+        nsfw_content_detected (`List[bool]`)
+            List of flags denoting whether the corresponding generated image likely represents "not-safe-for-work"
+            (nsfw) content.
+    """
+
+    images: Union[List[PIL.Image.Image], np.ndarray]
+    nsfw_content_detected: List[bool]
+
+
+@flax.struct.dataclass
+class InferenceState:
```

Comment:
this should be removed - let's just make it an inference state

Reply:
This could potentially be helpful to override pipeline modules, as in my code snippet above https://github.com/huggingface/diffusers/pull/559#issuecomment-1250830924.

We can do the same with a dictionary, but it's uglier in my opinion. Or with a helper function that returns a dict.

Reply:
For now I think it can just be a `dict` no? `dicts` are more universal and it means that not every pipeline has to have a data class state

Reply:
Removing for now -> let's maybe add later again if necessary